simpeg_drivers/joint/joint_petrophysics/driver.py

```diff
@@ -0,0 +1,234 @@
+
+    def __init__(self, params: JointPetrophysicsOptions):
+        self._wires = None
+        self._class_mapping: np.ndarray | None = None
+        self._directives = None
+        self._membership: np.ndarray = None
+        self._gaussian_model = None
+        self._pgi_regularization: PGIsmallness | None = None
+
+        super().__init__(params)
+
+        with fetch_active_workspace(self.workspace, mode="r+"):
+            self.initialize()
+
+    @property
+    def directives(self):
+        if getattr(self, "_directives", None) is None and not self.params.forward_only:
+            with fetch_active_workspace(self.workspace, mode="r+"):
+                directives_list = self._get_drivers_directives()
+                directives_list.append(
+                    directives.PGI_UpdateParameters(
+                        update_gmm=True,
+                        kappa=0.0,
+                        fixed_membership=np.c_[
+                            np.arange(self.models.n_active), self.membership
+                        ],
+                    )
+                )
+                directives_list += self._get_global_model_save_directives()
+                directives_list.append(
+                    directives.SavePGIModel(
+                        self.inversion_mesh.entity,
+                        self.pgi_regularization,
+                        self.geo_units,
+                        [driver.params.physical_property for driver in self.drivers],
+                        transforms=[
+                            lambda x: np.r_[list(self.geo_units.keys())][
+                                self.class_mapping
+                            ][x.astype(int)],
+                            maps.InjectActiveCells(
+                                self.inversion_mesh.mesh, self.models.active_cells, 0
+                            ),
+                        ],
+                        reference_type=self.params.petrophysics_model.entity_type,
+                    )
+                )
+                directives_list.append(
+                    directives.SaveLPModelGroup(
+                        self.inversion_mesh.entity,
+                        self._directives.update_irls_directive,
+                    )
+                )
+                directives_list.append(self._directives.save_iteration_log_files)
+                self._directives.directive_list = (
+                    self._directives.inversion_directives + directives_list
+                )
+        return self._directives
+
+    def get_regularization(self):
+        """
+        Create a flat ComboObjectiveFunction from all drivers provided and
+        add cross-gradient regularization for all combinations of model parameters.
+        """
+        regularizations = super().get_regularization()
+        reg_list, multipliers = self._overload_regularization(regularizations)
+        reg_list.append(self.pgi_regularization)
+        # TODO: Assign value from UIjson
```

Comment:
Is this still needed? self.params.alpha_s should contain data from the ui.json, no?